sphinx/screens-detail/payments/payment-template/payment-template/src/main/java/chat/sphinx/payment_template/ui/PaymentTemplateFragment.kt

```diff
@@ -0,0 +1,206 @@
+    }
+
+    override suspend fun onSideEffectCollect(sideEffect: PaymentTemplateSideEffect) {
+        sideEffect.execute(requireActivity())
+    }
+
+    override suspend fun onViewStateFlowCollect(viewState: PaymentTemplateViewState) {
+        when (viewState) {
+            is PaymentTemplateViewState.Idle -> {}
+
+            is PaymentTemplateViewState.ProcessingPayment -> {
+                binding.progressBarConfirm.visible
+            }
+            is PaymentTemplateViewState.PaymentFailed -> {
+                binding.progressBarConfirm.gone
+            }
+        }
+    }
+
+    override fun subscribeToViewStateFlow() {
+        super.subscribeToViewStateFlow()
+
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.templateImagesViewStateContainer.collect { viewState ->
+                @Exhaustive
+                when (viewState) {
+                    is TemplateImagesViewState.LoadingTemplateImages -> {
+                        binding.layoutConstraintProgressBarLoadingTemplates.visible
+                    }
+                    is TemplateImagesViewState.TemplateImages -> {
+                        binding.layoutConstraintProgressBarLoadingTemplates.gone
+
+                        binding.recyclerViewPaymentTemplate.apply {
+
+                            (adapter as PaymentTemplateAdapter).setItems(viewState.templates)
+
+                            //scrolling to first template item
+                            scrollBy(
+                                context.resources.getDimension(R.dimen.payment_template_recycler_view_item_width).toInt(),
+                                0
+                            )
+                        }
+                    }
+                }
+            }
+        }
+
+        onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+            viewModel.selectedTemplateViewStateContainer.collect { viewState ->
+                @Exhaustive
+                when (viewState) {
+                    is SelectedTemplateViewState.Idle -> {
+                        binding.imageViewSelectedPaymentTemplate.setImageDrawable(
+                            ContextCompat.getDrawable(binding.root.context, R.drawable.ic_no_template_with_padding)
+                        )
+                    }
+                    is SelectedTemplateViewState.SelectedTemplate -> {
+                        val template = viewState.template
+
+                        template.getTemplateUrl(MediaHost.DEFAULT.value)?.let { url ->
+                            val token = AuthenticationToken(template.token)
+
+                            imageLoader.load(
+                                binding.imageViewSelectedPaymentTemplate,
+                                url,
+                                ImageLoaderOptions.Builder()
+                                    .addHeader(token.headerKey, token.headerValue)
+                                    .build()
+                            )
```

Comment:
job and state need to be tracked in order to cancel disposable upon viewstate changes.